include/mixer.h

```diff
@@ -52,6 +52,14 @@ enum MixerModes {
 	M_16M,M_16S
 };
 
+enum MixerState : uint8_t {
+	Uninitialized = 1 << 0,
+	NoSound       = 1 << 1,
+	Off           = 1 << 2,
+	Mute          = 1 << 3,
+	On            = 1 << 4,
```

Comment:
If I'm not mistaken, you're not actually using `MixerState` as bitflags, so you could simplify this to become a regular enum.

Reply:
Removed :+1: 

One of the challenges is that we've got multiple "off-like" states that do the same thing. So going from Off to Mute, is still a state-change, but it's something we'd rather skip entirely.  

The new code uses a couple if/else's to compare the gross states as either "On" againt "Not-On", and only transition when needed.